sdk/core/azure-core/azure/core/pipeline/transport/_requests_basic.py

```diff
@@ -108,6 +130,19 @@ def __init__(self, pipeline, response):
         self.iter_content_func = self.response.internal_response.iter_content(self.block_size)
         self.content_length = int(response.headers.get('Content-Length', 0))
         self.downloaded = 0
+        headers = response.request.headers
+        if "x-ms-range" in headers:
+            self.range_header = "x-ms-range"
+            self.range = headers["x-ms-range"]
```

Comment:
A more robust to set `self.range` is the *response* headers, as this tells you which range the server actually responses with (keep in mind that the server can legitimately ignore the range header and return the full content!). See https://tools.ietf.org/html/rfc7233#section-4.1